api/osb/store_instances_plugin.go

```diff
@@ -0,0 +1,707 @@
+	}
+
+	if _, err := storage.Create(ctx, operation); err != nil {
+		return util.HandleStorageError(err, string(operation.GetType()))
+	}
+
+	return nil
+}
+
+func (ssi *StoreServiceInstancePlugin) storeInstance(ctx context.Context, storage storage.Repository, req *provisionRequest, ready bool) error {
+	planID, err := ssi.findPlanIDForInput(ctx, storage, req.BrokerID, req.ServiceID, req.PlanID)
+	if err != nil {
+		return err
+	}
+	instanceName := gjson.GetBytes(req.RawContext, "instance_name").String()
+	if len(instanceName) == 0 {
+		log.C(ctx).Debugf("Instance name missing. Defaulting to id %s", req.InstanceID)
+		instanceName = req.InstanceID
+	}
+	instance := &types.ServiceInstance{
+		Base: types.Base{
+			ID:        req.InstanceID,
+			CreatedAt: req.Timestamp,
+			UpdatedAt: req.Timestamp,
+			Labels:    make(map[string][]string),
+		},
+		Name:            instanceName,
+		ServicePlanID:   planID,
+		PlatformID:      req.PlatformID,
+		MaintenanceInfo: req.RawMaintenanceInfo,
+		Context:         req.RawContext,
+		Ready:           ready,
+		Usable:          true,
+	}
+	if _, err := storage.Create(ctx, instance); err != nil {
+		return util.HandleStorageError(err, string(instance.GetType()))
+	}
+	return nil
+}
+
+func (ssi *StoreServiceInstancePlugin) updateInstance(ctx context.Context, req *updateRequest, storage storage.Repository, usable bool) error {
+	byID := query.ByField(query.EqualsOperator, "id", req.InstanceID)
+	var instance types.Object
+	var err error
+	if instance, err = storage.Get(ctx, types.ServiceInstanceType, byID); err != nil {
+		if err != util.ErrNotFoundInStorage {
+			return util.HandleStorageError(err, string(types.ServiceInstanceType))
+		}
+	}
+	if instance == nil {
+		return nil
+	}
+	serviceInstance := instance.(*types.ServiceInstance)
+	oldServicePlanID := serviceInstance.ServicePlanID
+	previousValuesBytes, err := json.Marshal(req.PreviousValues)
+	if err != nil {
+		return err
+	}
+	previousValuesBytes, err = sjson.SetBytes(previousValuesBytes, smServicePlanIDKey, oldServicePlanID)
+	if err != nil {
+		return err
+	}
+	if len(req.PlanID) != 0 && req.PreviousValues.PlanID != req.PlanID {
+		var err error
+		serviceInstance.ServicePlanID, err = findServicePlanIDByCatalogIDs(ctx, storage, req.BrokerID, req.ServiceID, req.PlanID)
+		if err != nil {
+			return err
+		}
+	}
+	if len(req.MaintenanceInfo) != 0 {
+		serviceInstance.MaintenanceInfo = req.MaintenanceInfo
+	}
+	if len(req.RawContext) != 0 {
+		serviceInstance.Context = req.RawContext
+	}
+
+	serviceInstance.PreviousValues = previousValuesBytes
+	if _, err := storage.Update(ctx, serviceInstance, query.LabelChanges{}); err != nil {
+		return util.HandleStorageError(err, string(serviceInstance.GetType()))
+	}
+
+	return nil
+}
+
+func (ssi *StoreServiceInstancePlugin) rollbackInstance(ctx context.Context, req commonOSBRequest, storage storage.Repository, usable bool) error {
+	byID := query.ByField(query.EqualsOperator, "id", req.GetInstanceID())
```

Comment:
only `instanceID` is used from the req, so pass only the instanceID to the method

Reply:
currently request is also used on line 593